precovery/frame_db.py

```diff
@@ -626,9 +620,66 @@ def load_csv(
                 data_offset=offset,
                 data_length=length,
             )
-            frames_to_add.append(frame)
 
-        self.idx.add_frames(frames_to_add)
+            healpix_frames.append(frame)
+
+        self.idx.add_frames(healpix_frames)
+
+    @staticmethod
+    def _compute_year_month_str(frame: sourcecatalog.SourceFrame) -> str:
+        """Gets the Year and Month parts of a SourceFrame's exposure
+        midpoint timestamp, and returns them in "YYYY-MM" format.
+
+        This is used as a partitioning key for observation data files.
+
+        """
+        time = Time(frame.exposure_mjd_mid, format="mjd", scale="utc")
+        return time.strftime("%Y-%m")
+
+    def load_csv(
+        self,
+        csv_file: str,
+        dataset_id: str,
+        skip: int = 0,
+        limit: Optional[int] = None,
+        name: Optional[str] = None,
+        reference_doi: Optional[str] = None,
```

Comment:
You know, I think even with this utility it's totally reasonable to split out the `add_dataset` function. If we want to index files from the cli, our cli function can take the command line arguments and make separate calls to `add_dataset` and `load_csv`. If we're calling these methods programmatically (say, inside an ETL process), then it's just as good to be able to make separate calls. That way, the simplest case remains simple (loading a single file from cli) but we don't get this weird thing where loading 20 csv's for a large dataset tries to upsert the dataset info 20 times.

Reply:
Ah, so you're suggesting we drop the `add_dataset` call from in here? That does make the function really small, which is pretty nice. 

Does `load_csv` have any callers? I don't know what kind of scripts and automation we have that might be depending on its present signature.